flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -641,6 +648,33 @@ public void cancel() {
         return executionGraph.getTerminationFuture().thenApply(FunctionUtils.nullFn());
     }
 
+    protected void archiveGlobalFailure(Throwable failure) {
+        Set<ExecutionAttemptID> executionAttemptIds =
+                IterableUtils.toStream(executionGraph.getAllExecutionVertices())
+                        .map(ExecutionVertex::getCurrentExecutionAttempt)
+                        .map(Execution::getAttemptId)
+                        .collect(Collectors.toSet());
+
+        globalTaskFailures.put(
+                executionAttemptIds, new ErrorInfo(failure, System.currentTimeMillis()));
+
+        log.debug("Archive global {}: {}", failure, failure.getMessage());
```

Comment:
How would this log line look like? Wouldn't `failure.getMessage` be duplicated? Maybe also say "Archive global failure...".